zio-http/src/main/scala/zhttp/http/Headers.scala

```diff
@@ -21,6 +21,8 @@ final case class Headers(toChunk: Chunk[Header]) extends HeaderExtension[Headers
 
   override def updateHeaders(f: Headers => Headers): Headers = f(self)
 
+  def filter(f: Header => Boolean): Headers = Headers(self.toChunk.filter(f(_)))
```

Comment:
- We can add this to `HeadersModifiers`
- For this PR we don't need this function it. You could have use `removeHeader(Name.ContentLength)`